api/lib/domain/usecases/compute-campaign-analysis.js

```diff
@@ -0,0 +1,52 @@
+const _ = require('lodash');
+const CampaignAnalysis = require('../models/CampaignAnalysis');
+const CampaignTubeRecommendation = require('../models/CampaignTubeRecommendation');
+const { UserNotAuthorizedToAccessEntity } = require('../errors');
+
+module.exports = async function computeCampaignAnalysis(
+  {
+    userId,
+    campaignId,
+    campaignRepository,
+    competenceRepository,
+    targetProfileRepository,
+    tubeRepository,
+  } = {}) {
+
+  const hasUserAccessToResult = await campaignRepository.checkIfUserOrganizationHasAccessToCampaign(campaignId, userId);
+
+  if (!hasUserAccessToResult) {
+    throw new UserNotAuthorizedToAccessEntity('User does not have access to this campaign');
+  }
+
+  const competences = await competenceRepository.list();
```

Comment:
si on souhaite résoudre les 3 promesses en même temps, on peut faire:
```suggestion
  const [competences, tubes, targetProfile] = await Promise.all([
   competenceRepository.list(),
   tubeRepository.list(),
   targetProfileRepository.getByCampaignId(campaignId)
]);
```